sympy/integrals/manualintegrate.py

```diff
@@ -390,6 +388,10 @@ def pull_out_u_rl(integrand):
             u = u.subs(dummy, 1)
             dv = dv.subs(dummy, 1)
 
+            # Don't pick a non-polynomial algebraic to be differentiated
+            if rule == pull_out_algebraic and not u.is_polynomial(symbol):
```

Comment:
Is there some reason why this polynomial test could not be in `pull_out_algebraic`?

Reply:
Yes. As currently implemented, integration by parts is done only if the integrand is the product of two LIATE terms. This is checked in the loop just below, line 395. After some u, dv are found, this loop goes through the remaining LIATE rules looking for one that could pick this dv as its u.  

So, if we prohibit picking non-polynomial algebraic terms in `pull_out_algebraic`, then they won't be picked as dv either.  For example, sqrt(x)*log(x) will not be integrated by parts, because even though log(x) qualifies as u, there would be no rule picking sqrt(x) as u.

Reply:
Now that I looked at that line again, `sympy.simplify(r[0].subs(dummy, 1)) == sympy.simplify(dv)` looks like it should have been `r[0].subs(dummy, 1).equals(dv)` since the goal is to test equality, not to simplify both things. I'll try that.

Reply:
Hmm... Testing with this patch
```
diff --git a/sympy/integrals/manualintegrate.py b/sympy/integrals/manualintegrate.py
index 7db7dff..008c8f1 100644
--- a/sympy/integrals/manualintegrate.py
+++ b/sympy/integrals/manualintegrate.py
@@ -351,7 +351,7 @@ def pull_out_algebraic(integrand):
         if algebraic:
             u = sympy.Mul(*algebraic)
             dv = (integrand / u).cancel()
-            if not u.is_polynomial() and isinstance(dv, sympy.exp):
+            if not u.is_polynomial(symbol):
                 return
             return u, dv
```
I get
```
>>> manualintegrate.manualintegrate(sqrt(x)*log(x), x)
2*x**(3/2)*log(x)/3 - 4*x**(3/2)/9
>>> manualintegrate.manualintegrate((1+sin(x))/x, x)
log(x) + Integral(sin(x)/x, x)
```
which looks OK to me. Am I missing something?

Reply:
The test that actually failed for me when I tried that was [this one](https://github.com/sympy/sympy/blob/master/sympy/integrals/tests/test_manual.py#L326), `manualintegrate(x**Rational(3,2) * log(x), x) `

Reply:
Now I got some idea of how this is working, thanks.